packages/components/src/components/user/CreateUsersModal.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react'
+import { Checkbox, FormControl, Modal } from "react-bootstrap";
+import { Security } from "@labkey/api";
+import { WizardNavButtons } from "../buttons/WizardNavButtons";
+import { Alert } from "../base/Alert";
+import { SelectInput } from "../forms/input/SelectInput";
+
+interface Props {
+    onCancel: () => void
+    onComplete: (response: any, role: string) => void
+    show: boolean
+
+    // optional array of role options, objects with id and label values
+    // note that the createNewUser action will not use this value but it will be passed back to the onComplete
+    roleOptions?: Array<any>
+}
+
+interface State {
+    emailText: string
+    sendEmail: boolean
+    role: string
+    isSubmitting: boolean
+    error: string
+}
+
+const DEFAULT_STATE = {
+    emailText: '',
+    sendEmail: true,
+    role: undefined,
+    isSubmitting: false,
+    error: undefined
+};
+
+export class CreateUsersModal extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = DEFAULT_STATE
+    }
+
+    handleEmail = (evt: any) => {
+        const emailText = evt.target.value;
+        this.setState(() => ({emailText}));
+    };
+
+    handleSendEmail = (evt: any) => {
+        const sendEmail = evt.target.checked;
+        this.setState(() => ({sendEmail}));
+    };
+
+    handleRole = (name, formValue, selectedOption) => {
+        const role = selectedOption ? selectedOption.id : undefined;
+        this.setState(() => ({role}));
+    };
+
+    createUsers = () => {
+        const { emailText, sendEmail } = this.state;
+        this.setState(() => ({isSubmitting: true, error: undefined}));
+
+        // convert the email addresses from newline separated to semicolon separated
+        const email = emailText.split('\n').join(';');
```

Comment:
Maybe use a replace-all here to avoid creation of strings?  (minor optimization)

Reply:
Done.